include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -754,6 +758,74 @@ __select_backend(const execution::fpga_policy<_Factor, _KernelName>&, _Ranges&&.
 }
 #endif
 
+// Check the outer view type type to see if we can vectorize. Any non-contiguous inputs (e.g. reverse
+// views, permutation views, etc.) cannot be vectorized. If C++20 ranges are present, then we can
+// use the std::ranges::contiguous_range concept.
+template <typename _Rng>
+struct __is_vectorizable_range
+{
+    constexpr static bool value =
+#if _ONEDPL_CPP20_RANGES_PRESENT && _ONEDPL_CPP20_CONCEPTS_PRESENT
+        std::ranges::contiguous_range<_Rng> ||
+#endif
+        __nanorange::nano::ranges::contiguous_range<_Rng>;
+};
+
+// Guard view specializations
+template <typename _Type>
+struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<_Type*>> : std::true_type
+{
+};
+
+// Counting iterator does not go through global memory but does not disable vectorization elsewhere.
+template <typename _Ip>
+struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<oneapi::dpl::counting_iterator<_Ip>>> : std::true_type
+{
+};
+
+// Discard iterator does not go through global memory but does not disable vectorization elsewhere.
+template <>
+struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<oneapi::dpl::discard_iterator>> : std::true_type
+{
+};
+
+// For any other iterator over a guard_view, use contiguous iterator concepts
+template <typename _Iterator>
+struct __is_vectorizable_range<oneapi::dpl::__ranges::guard_view<_Iterator>> : std::true_type
```

Comment:
I am afraid such definition(specialization)  of `__is_vectorizable_range` trait is not correct.
F.e.
`bool bres =  __is_vectorizable_range{};` is identically true, because there is `operator value_type()` which which is inherited from base type - `std::true_type`.

Reply:
Thanks, this is a typo, and I have fixed it. Inheriting std::true_type should not be there.